developer_guides/testbench/debug_in_testbench.rst

```diff
@@ -0,0 +1,171 @@
+binary. The arguments specify e.g. the input and output sample rate,
+input and output RAW data files, topology to use for testing and
+sample format. The command line options are described when invoking
+the binary with switch "-h". But for the binary to work correctly the
+dynamic libaries path need to be instructed for the operating
+system. It is done by setting the environment variable
+LD_LIBRARY_PATH to above shown value.
+
+.. code-block:: bash
+
+   export LD_LIBRARY_PATH=../../testbench/build_testbench/sof_ep/install/lib:../../testbench/build_testbench/sof_parser/install/lib
+   ../../testbench/build_testbench/install/bin/testbench -h
+
+If the help text appeared the testbench binary start directly from command line worked. Next, the testbench can
+started in ddd debugger. It and the dependencies such as gdb needs to be installed if missing.
+
+.. code-block:: bash
+
+   sudo apt install ddd
+
+The debugging is started to previously used shell with LD_LIBRARY_PATH set.
+
+.. code-block:: bash
+
+   ddd ../../testbench/build_testbench/install/bin/testbench
+
+This opens the debugger window. From there find the code line just
+after topology parsing (currently 295) with mouse scroll in the code
+window and place a break point there with right mouse button (a red
+stop sign). If there are issues those happen at topology parsing or
+ẃithin component in instantiation in new() place the breakpoint to
+parse_topology() line.
+
+.. figure:: fig_ddd.png
+
+	    The ddd debugger start view.
+
+.. figure:: fig_add_breakpoint.png
+
+	    Breakpoint added with right mouse click.
+	    
+The breakpoint is placed after topology parsing since the component
+symbols do not exist in debugger context before it is loaded by the
+topology. To run the testbench until breakpoint, select from menu
+Program -> Run... Then mouse copy (text select with left button and
+enter to field with center button) the argument line output from
+previous script run and click "Run".
+
+::
+
+   -d -r 48000 -R 48000 -i audio_in.raw -o audio_out.raw -t ../../build_tools/test/topology/test-playback-ssp5-mclk-0-I2S-eq-iir-s16le-s16le-48k-24576k-codec.tplg -b S16_L
+
+The execution is now stopped to breakpoint. Since the symbols exist
+now the breakpoints can be added to component life cycle after
+new(). Use the lowest window parth with prompt (gdb) for convenience.
+
+.. code-block:: bash
+
+   break eq_iir_cmd
+   break eq_iir_params
+   break eq_iir_prepare
+   break eq_iir_copy
+   break eq_iir_reset
+   break eq_iir_free
+
+After that press "Cont" in the small remote control window next to
+main ddd window. The execution stops to params() function in playback
+start. To view stream parameters mouse left click on params in
+function arguments list and select with right mouse button "Display
+\*params". The same can be done for dev structure. The not shown fiels
+can be expanded and pointers should as pipeline from dev can be looked
```

Comment:
be expanded and pointers should as pipeline from dev can be -> something wrong here :)

Reply:
I improved it, hope it's OK now!